samples/pyarrow/append_rows_with_arrow.py

```diff
@@ -0,0 +1,206 @@
+
+
+def generate_write_requests(num_rows=TABLE_LENGTH):
+    date_1 = datetime.date(2020, 10, 1)
+    date_2 = datetime.date(2021, 10, 1)
+
+    datetime_1 = datetime.datetime(2016, 12, 3, 14, 11, 27, 123456)
+    datetime_2 = datetime.datetime(2017, 12, 3, 14, 11, 27, 123456)
+
+    timestamp_1 = datetime.datetime(
+        1999, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+    timestamp_2 = datetime.datetime(
+        2000, 12, 31, 23, 59, 59, 999999, tzinfo=datetime.timezone.utc
+    )
+
+    # Pandas Dataframe.
+    rows = []
+    for i in range(num_rows):
+        row = {
+            "bool_col": True,
+            "int64_col": i,
+            "float64_col": float(i),
+            "numeric_col": decimal.Decimal("0.000000001"),
+            "bignumeric_col": decimal.Decimal("0.1234567891"),
+            "string_col": "data as string",
+            "bytes_col": str.encode("data in bytes"),
+            "date_col": datetime.date(2019, 5, 10),
+            "datetime_col": datetime_1,
+            "time_col": datetime.time(23, 59, 59, 999999),
+            "timestamp_col": timestamp_1,
+            "geography_col": "POINT(-121 41)",
+            "range_date_col": {"start": date_1, "end": date_2},
+            "range_datetime_col": {"start": datetime_1, "end": datetime_2},
+            "range_timestamp_col": {"start": timestamp_1, "end": timestamp_2},
+        }
+        rows.append(row)
+    df = pd.DataFrame(rows)
+
+    # Dataframe to PyArrow Table.
+    table = pa.Table.from_pandas(df, schema=PYARROW_SCHEMA)
+
+    # Determine max_chunksize of the record batches. Because max size of
+    # AppendRowsRequest is 10 MB, we need to split the table if it's too big.
+    # See: https://github.com/googleapis/googleapis/blob/27296636cf8797026124cd67034b42190ab602a4/google/cloud/bigquery/storage/v1/storage.proto#L422
+    max_request_bytes = 10 * 2**20  # 10 MB
+    chunk_num = int(table.nbytes / max_request_bytes) + 1
+    chunk_size = int(table.num_rows / chunk_num)
+
+    # Construct request(s).
+    for batch in table.to_batches(max_chunksize=chunk_size):
+        request = gapic_types.AppendRowsRequest()
+        request.arrow_rows.rows.serialized_record_batch = batch.serialize().to_pybytes()
+        yield request
+
+
+def append_rows(bqstorage_write_client, table):
+    append_rows_stream = create_stream(bqstorage_write_client, table)
+    futures = []
+
+    for request in generate_write_requests(num_rows=TABLE_LENGTH):
+        response_future = append_rows_stream.send(request)
+        futures.append(response_future)
+        response_future.result()
+
+    return futures
+
+
+def verify_result(client, table, futures):
+    bq_table = client.get_table(table)
+
+    # Verify table schema.
+    assert bq_table.schema == BQ_SCHEMA
+
+    # Verify table size.
+    query = client.query(f"SELECT COUNT(1) FROM `{bq_table}`;")
+    query_result = query.result().to_dataframe()
+    assert query_result.iloc[0, 0] == TABLE_LENGTH
```

Comment:
To account for retries / at-least-once, you might want to relax this check to greater or equal.

Reply:
That's a great idea! done